stacks-signer/src/client/stacks_client.rs

```diff
@@ -220,17 +434,16 @@ impl StacksClient {
         Ok(peer_info.burn_block_height)
     }
 
-    /// Helper function to retrieve the current reward cycle number from the stacks node
+    /// Get the current reward cycle from the stacks node
     pub fn get_current_reward_cycle(&self) -> Result<u64, ClientError> {
```

Comment:
The current cycle is directly available from /v2/pox -- it's under `.current_cycle.id`

Reply:
It isn't exactly -- because `/v2/pox` performs a clarity read-only call, its evaluated at the _stacks tip_, where `burn-height` may not but the burn tip (in fact, it isn't _ever_ in Stacks 2.x). This means that `/v2/pox` will be one cycle off on the 0 block of a cycle. This is a problem for signers because they need to sign that block.

Reply:
I think that the `/v2/pox` endpoint would be improved by something like this:

```diff
diff --git a/stackslib/src/net/api/getpoxinfo.rs b/stackslib/src/net/api/getpoxinfo.rs
index d476d27a6..befc19406 100644
--- a/stackslib/src/net/api/getpoxinfo.rs
+++ b/stackslib/src/net/api/getpoxinfo.rs
@@ -202,12 +202,6 @@ impl RPCPoxInfoData {
             .to_owned()
             .expect_u128()? as u64;
 
-        let reward_cycle_id = res
-            .get("reward-cycle-id")
-            .expect(&format!("FATAL: no 'reward-cycle-id'"))
-            .to_owned()
-            .expect_u128()? as u64;
-
         let reward_cycle_length = res
             .get("reward-cycle-length")
             .expect(&format!("FATAL: no 'reward-cycle-length'"))
@@ -268,7 +262,13 @@ impl RPCPoxInfoData {
             return Err(NetError::DBError(DBError::Corruption));
         }
 
+        let reward_cycle_id = burnchain.block_height_to_reward_cycle(
+            burnchain_tip.block_height
+        ).ok_or_else(|| {
+            NetError::ChainstateError("Current burn block height is before stacks start".into())
+        })?;
         let effective_height = burnchain_tip.block_height - first_burnchain_block_height;
+
         let next_reward_cycle_in = reward_cycle_length - (effective_height % reward_cycle_length);
 
         let next_rewards_start = burnchain_tip.block_height + next_reward_cycle_in;
```

Then it wouldn't suffer from this problem. But such a change should be tested, and I don't think this PR should incorporate that change or be blocked by it. So, I suggested calculating the current cycle manually with the information provided by `/v2/pox` that doesn't depend on the read-only call.